File: contrib/playlib/api/src/RouteCompilerWorkerApi.scala
package mill
package playlib
package api

import ammonite.ops.Path
import mill.api.Result
import mill.scalalib.api.CompilationResult


trait RouteCompilerWorkerApi {
  def compile(files: Seq[Path],
              additionalImports: Seq[String],
              forwardsRouter: Boolean,
              reverseRouter: Boolean,
              namespaceReverseRouter: Boolean,
              generatorType:RouteCompilerType,
              dest: Path)(implicit ctx: mill.api.Ctx):Result[CompilationResult]


}

sealed trait RouteCompilerType
object RouteCompilerType{
  case object InjectedGenerator extends RouteCompilerType
  case object StaticGenerator extends RouteCompilerType
}

File: contrib/playlib/api/src/RouteCompilerWorkerApi.java
package mill.playlib.api;
public interface RouteCompilerWorkerApi {
    String compile(java.io.File[] files,
                   String[] additionalImports,
                   boolean forwardsRouter,
                   boolean reverseRouter,
                   boolean namespaceReverseRouter,
                   RouteCompilerType generatorType,
                   java.io.File dest);


}
